lib/pbench/agent/utils.py

```diff
@@ -390,3 +393,60 @@ def collect_local_info(pbench_bin):
         hostdata[arg] = cp.stdout.strip() if cp.stdout is not None else ""
 
     return (version, seqno, sha1, hostdata)
+
+
+if os.environ.get("_PBENCH_UNIT_TESTS"):
+
+    class LocalRemoteHost:
+        """Mock out behavior for legacy unit tests.
+        """
+
+        def __init__(self):
+            self._local_names = frozenset(
+                [
+                    "127.0.0.1",
+                    "localhost",
+                    "172.16.42.42",
+                    "testhost.example.com",
+                    "testhost",
+                ]
+            )
+
+        def is_remote(self, host_name):
+            return host_name not in self._local_names
+
+
+else:
+
+    class LocalRemoteHost:
+        """Dynamically build a list of all the local IP addresses so that a simple
+        simple query interface provides the answer to the question, "is the given
+        host name or IP address remote?"
+        """
+
+        def __init__(self):
+            """Build up a list of local IP addresses from all the interfaces.
+            """
+            local_ips_l = []
+            for adapter in ifaddr.get_adapters():
+                for ip in adapter.ips:
+                    local_ips_l.append(ip.ip[0] if isinstance(ip.ip, tuple) else ip.ip)
+            self.local_ips = frozenset(local_ips_l)
+
+        def is_remote(self, host_name):
+            """Returns True if the given host name (or IP address) is remote else
+            returns False.
+            """
+            ips_l = []
+            try:
+                ip_addrs = socket.getaddrinfo(host_name, None)
+                for addrinfo in ip_addrs:
+                    ips_l.append(addrinfo[4][0])
+            except socket.gaierror:
+                try:
+                    ipaddress.ip_address(host_name)
+                except ValueError:
+                    return True
+                ips_l.append(host_name)
+            ips = frozenset(ips_l)
```

Comment:
There's no need to do an append operation -- we can just assign a new one to `ips_l`, and while we're at it, we can skip the middleman and make it a frozenset right off the bat and skip the assignment at line 451 if we do the analogous thing at line 443/444:
```
                ips = frozenset({host_name})
```